templates/solochain/pallets/template/src/mock.rs

```diff
@@ -4,14 +4,30 @@ use sp_runtime::BuildStorage;
 
 type Block = frame_system::mocking::MockBlock<Test>;
 
-// Configure a mock runtime to test the pallet.
-frame_support::construct_runtime!(
-	pub enum Test
-	{
-		System: frame_system,
-		TemplateModule: pallet_template,
-	}
-);
+#[frame_support::runtime]
+mod runtime {
+	// The main runtime
+	#[runtime::runtime]
+	// Runtime Types to be generated
+	#[runtime::derive(
+		RuntimeCall,
+		RuntimeEvent,
+		RuntimeError,
+		RuntimeOrigin,
+		RuntimeFreezeReason,
+		RuntimeHoldReason,
+		RuntimeSlashReason,
+		RuntimeLockId,
+		RuntimeTask
+	)]
```

Comment:
not sure the philosophy we want to follow, but the template does not need all these derives. i could see an argument to use it either way

Reply:
I'm pro reducing this, but I cannot find the documentation explaining what each does. My personal preference would be to exclude them, and explain in the template why we are excluding them.

Reply:
if the type isnt being used anywhere in the file, you can pretty safely delete it.

Reply:
@Jan-Jan seems like the PR got merged without this being addressed. Looks like a good follow up. You can also look at the `mock.rs` in the parachain template and fix this there too.